bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/IndexPersistenceMgr.java

```diff
@@ -354,8 +316,7 @@ private void getActiveLedgers() throws IOException {
      */
     void removeLedger(Long ledgerId) throws IOException {
         // Delete the ledger's index file and close the FileInfo
-        FileInfo fi = null;
-        fileInfoLock.writeLock().lock();
```

Comment:
Remove ledger is not synchronized with getFileInfo, so it will introduce race conditions.

Race Condition:
1. Thread A is on line 221 and get a FileInfo from cache successfully, but haven't increment the refCount.
2. Thread B is on line 335 and after it invalidate the FileInfo from cache, refCount will becomes 0 so the underline FileChannel will be closed.
3. Now Thread A continues to increment the refCount, and returns the closed FileInfo to application.
4. java.nio.channels.ClosedChannelException will be thrown when we starts to use this FileInfo.

Reply:
This is same race as https://github.com/apache/bookkeeper/pull/913#discussion_r158849128, no?

Reply:
Yeah, similar but these race conditions caused a lot of issues to our production cluster under high load, we've been testing these for months.